src/main/java/com/google/api/codegen/configgen/transformer/MethodTransformer.java

```diff
@@ -142,6 +162,9 @@ private PageStreamingResponseView generatePageStreamingResponse(Method method) {
               method.getSimpleName());
           break;
         }
+      } else if (field.getApiSource().equals(ApiSource.DISCOVERY)) {
```

Comment:
Is there any way to base this off of data from pagingParameters instead of making a discovery-specific check?

Reply:
Made this MethodTransformer class an abstract class that can do the bulk of the transformations but leaves it up to subclasses to do ApiSource-specific work.